src/back/controllers/api/routes/categories/categories.spec.js

```diff
@@ -23,22 +24,50 @@ describe('categories handler', () => {
         });
     });
 
-    it('should return all data from model', () => {
-        sut.getAll();
-        categoryModel.find.should.have.been.calledWith({});
-    });
+    describe('getAll', () => {
+        let res;
+        let req;
+        let data;
+        let userData;
+        let onCategoryFindCallback;
+        let onUserFindCallback;
+        let expectedData;
 
-    it('should return data as parsed json', () => {
-        const res = { json: env.stub() };
-        const data = 'data';
+        beforeEach(() => {
+            expectedData = [
+                { _id: 1, name: 'name1', checked: true },
+                { _id: 2, name: 'name2' }
+            ];
+            res = {
+                json: env.stub(),
+                sendStatus: env.stub()
+            };
+            req = {
+                user: { _id: 1 }
+            };
+            data = [{ _id: 1, name: 'name1' }, { _id: 2, name: 'name2' }];
+            userData = {
+                categories: [1]
+            };
+
+            sut.getAll(req, res);
+
+            onCategoryFindCallback = categoryModel.find.lastCall.args[1];
 
-        sut.getAll({}, res);
+            onCategoryFindCallback({}, data);
 
-        const onFindCallback = categoryModel.find.lastCall.args[1];
+            onUserFindCallback = userModel.findById.lastCall.args[1];
 
-        onFindCallback({}, data);
+            onUserFindCallback({}, userData);
+        });
+
+        it('should return all data from model', () => {
```

Comment:
Please keep relevant description for assertations you make, and
do not use low level implementation details in description for spec, code is sef-descriptive